src/parser/waypoint.rs

```diff
@@ -68,7 +68,10 @@ pub fn consume<R: Read>(context: &mut Context<R>, tagname: &'static str) -> GpxR
                 match name.local_name.as_ref() {
                     "ele" => {
                         // Cast the elevation to an f64, from a string.
-                        waypoint.elevation = Some(string::consume(context, "ele", false)?.parse()?)
+                        waypoint.elevation = match string::consume(context, "ele", false) {
+                            Ok(v) => Some(v.parse()?),
+                            Err(_) => None,
```

Comment:
Aren't there other errors that we'd want to surface here? I think the only error we'd want to map to `None` is `GpxError::NoStringContent`

```
Err(GpxError::NoStringContent) => None,
Err(other_err) => return Err(other_err)
```

For example, I think we'd want an element like this to continue to error, not return None:
```
<ele>1.asdf</ele>
```